python/rmm/_lib/device_buffer.pyx

```diff
@@ -92,39 +97,47 @@ cdef class DeviceBuffer:
         return buf
 
     @staticmethod
-    @cython.boundscheck(False)
-    cdef DeviceBuffer c_frombytes(const unsigned char[::1] b,
+    cdef DeviceBuffer c_to_device(const unsigned char[::1] b,
                                   uintptr_t stream=0):
-        if b is None:
-            raise TypeError(
-                "Argument 'b' has incorrect type"
-                " (expected bytes, got NoneType)"
+        return to_device(b, stream)
+
+    @staticmethod
+    def to_device(const unsigned char[::1] b, uintptr_t stream=0):
+        return to_device(b, stream)
+
+    cpdef copy_to_host(self, unsigned char[::1] hb=None, uintptr_t stream=0):
+        cdef const device_buffer* dbp = self.c_obj.get()
+        cdef size_t s = dbp.size()
+
+        if hb is None:
+            # NumPy leverages huge pages under-the-hood,
+            # which speeds up the copy from device to host.
+            hb = np.empty((s,), dtype="u1")
+        elif len(hb) < s:
+            raise ValueError(
+                "Argument `hb` is too small. Need space for %i bytes." % s
+            )
+        elif len(hb) > s:
+            hb = hb[:s]
+            warnings.warn(
+                "Argument `hb` larger than needed."
+                " Will fill only first %i bytes." % s,
+                RuntimeWarning
             )
 
-        cdef uintptr_t p = <uintptr_t>&b[0]
-        cdef size_t s = len(b)
-        return DeviceBuffer(ptr=p, size=s, stream=stream)
+        with nogil:
+            copy_to_host(<uintptr_t>dbp.data(), hb, stream)
```

Comment:
This name clashes with the member method of `DeviceBuffer` and makes things confusing, could we rename the free function (to keep this ducktyping)?

Reply:
Sure what do you think we should name it? `copy_ptr_to_host`?

Reply:
Have renamed to `copy_ptr_to_host` 🙂